### PR TITLE
feat(marketplace): `SellOrders` query & sub-queries

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- adds all SellOrders queries

ref: #856 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/marketplace/query_sell_orders.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	marketplacev1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	"github.com/regen-network/regen-ledger/types/ormutil"
+	v1 "github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
```
Comment (ryanchristo) [inline x/ecocredit/server/marketplace/query_sell_orders.go:12]: `api` and `marketplace`? Would be easier to maintain and no need to specify `v1`.

File: x/ecocredit/server/marketplace/query_sell_orders_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	"gotest.tools/v3/assert"
+
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	ecocreditv1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
```
Comment (ryanchristo) [inline x/ecocredit/server/marketplace/query_sell_orders_test.go:16]: `ecocreditapi` maybe?

File: x/ecocredit/server/marketplace/query_sell_orders_test.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	_, _, addr2 := testdata.KeyTestPubAddr()
+
+	insertSellOrder(t, s, s.addr, 1)
+	insertSellOrder(t, s, addr2, 1)
+
+	res, err := s.k.SellOrders(s.ctx, &marketplace.QuerySellOrdersRequest{
+		Pagination: nil,
+	})
+	assert.NilError(t, err)
+	assert.Equal(t, 2, len(res.SellOrders))
```
Comment (ryanchristo) [inline x/ecocredit/server/marketplace/query_sell_orders_test.go:40]: Should we also include an test case with pagination?

**Review by ryanchristo — APPROVED** [review]
Looks good to me. A couple nits with import names and maybe a test case using pagination would be nice. Pre-approving.

File: x/ecocredit/server/marketplace/query_sell_orders.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	pg, err := ormutil.GogoPageReqToPulsarPageReq(req.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := k.stateStore.SellOrderTable().List(ctx, api.SellOrderSellerIndexKey{}, ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
+	defer it.Close()
```
Comment (aleem1314) [inline x/ecocredit/server/marketplace/query_sell_orders.go:26]:
```suggestion
```


We are already closing iterator in [ormlist.Paginate](https://github.com/regen-network/regen-ledger/blob/dcc5dc72ce40d75e45323663b949fcd3629c7b4c/orm/iterator.go#L146).

File: x/ecocredit/server/marketplace/query_sell_orders.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+	batch, err := k.coreStore.BatchInfoTable().GetByBatchDenom(ctx, req.BatchDenom)
+	if err != nil {
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("could not get batch with denom %s: %s", req.BatchDenom, err.Error())
+	}
+
+	it, err := k.stateStore.SellOrderTable().List(ctx, api.SellOrderBatchIdIndexKey{}.WithBatchId(batch.Id), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
+	defer it.Close()
```
Comment (aleem1314) [inline x/ecocredit/server/marketplace/query_sell_orders.go:66]:
```suggestion
```

File: x/ecocredit/server/marketplace/query_sell_orders.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+	seller, err := sdk.AccAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := k.stateStore.SellOrderTable().List(ctx, api.SellOrderSellerIndexKey{}.WithSeller(seller), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
+	defer it.Close()
```
Comment (aleem1314) [inline x/ecocredit/server/marketplace/query_sell_orders.go:108]:
```suggestion
```

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**